.github/workflows/build-and-push-async-upload.yml

```diff
@@ -64,18 +64,24 @@ jobs:
 
     - name: Tag Latest
       # this workflow works on main branch and tags, so we need to tag as 'latest' and 'main'
-      shell: bash
-      env:
-        IMG: "${{ env.IMG_REGISTRY }}/${{ env.IMG_ORG }}/${{ env.IMG_NAME }}"
-      run: |
-        docker tag ${{ env.IMG }}:$VERSION ${{ env.IMG }}:latest
-        docker push ${{ env.IMG }}:latest
+      uses: docker/build-push-action@v6
+      with:
+        context: ./jobs/async-upload
+        push: true
+        tags: ${{ env.IMG_REGISTRY }}/${{ env.IMG_ORG }}/${{ env.IMG_NAME }}:latest
+        cache-from: |
+          type=gha
+          type=registry,ref=${{ env.IMG_REGISTRY }}/${{ env.IMG_ORG }}/${{ env.IMG_NAME }}:${{ env.VERSION }}
+        cache-to: type=gha,mode=max
 
     - name: Tag Main
       # this workflow works on main branch and tags, so we need to tag as 'latest' and 'main'
-      shell: bash
-      env:
-        IMG: "${{ env.IMG_REGISTRY }}/${{ env.IMG_ORG }}/${{ env.IMG_NAME }}"
-      run: |
-        docker tag ${{ env.IMG }}:$VERSION ${{ env.IMG }}:main
-        docker push ${{ env.IMG }}:main
+      uses: docker/build-push-action@v6
+      with:
+        context: ./jobs/async-upload
+        push: true
+        tags: ${{ env.IMG_REGISTRY }}/${{ env.IMG_ORG }}/${{ env.IMG_NAME }}:main
+        cache-from: |
+          type=gha
+          type=registry,ref=${{ env.IMG_REGISTRY }}/${{ env.IMG_ORG }}/${{ env.IMG_NAME }}:${{ env.VERSION }}
+        cache-to: type=gha,mode=max
```

Comment:
Can't we compact those three steps into one using:

```yaml
          tags: |
            ${{ env.IMG_REGISTRY }}/${{ env.IMG_ORG }}/${{ env.IMG_NAME }}:${{ env.VERSION }}
            ${{ env.IMG_REGISTRY }}/${{ env.IMG_ORG }}/${{ env.IMG_NAME }}:latest
            ${{ env.IMG_REGISTRY }}/${{ env.IMG_ORG }}/${{ env.IMG_NAME }}:main
```

Reply:
yeah that's a good point... let's try that (reworking PR)